File: auction/src/main/java/com/application/auction/dto/response/ProfileResponse.java
package com.application.auction.dto.response;

import com.application.auction.enums.KycStatus;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ProfileResponse {
    UUID userId;
    String fullName;
    String email;
    String phoneNumber;
    String avatar;
    String bio;
    Boolean isWalletActive;
    KycStatus kycStatus;
    String preferences;
}

File: app/lib/features/setting/data/models/profile_response.dart
class ProfileResponse {
  final String userId;
  final String? fullName;
  final String email;
  final String? phoneNumber;
  final String? avatar;
  final String? bio;
  final bool isWalletActive;
  final String kycStatus;
  final String preferences;

  ProfileResponse({
    required this.userId,
    required this.email,
    required this.isWalletActive,
    required this.kycStatus,
    required this.preferences,
    this.fullName,
    this.phoneNumber,
    this.avatar,
    this.bio,
  });

  factory ProfileResponse.fromJson(Map<String, dynamic> json) {
    return ProfileResponse(
      userId: json['userId']?.toString() ?? '',
      fullName: json['fullName']?.toString(),
      email: json['email']?.toString() ?? '',
      phoneNumber: json['phoneNumber']?.toString(),
      avatar: json['avatar']?.toString(),
      bio: json['bio']?.toString(),
      isWalletActive: json['isWalletActive'] == true,
      kycStatus: json['kycStatus']?.toString() ?? 'PENDING',
      preferences: json['preferences']?.toString() ?? '{}',
    );
  }
}
